src/geo/lng_lat.js

```diff
@@ -121,6 +144,13 @@ class LngLat {
             new LngLat(this.lng + lngAccuracy, this.lat + latAccuracy));
     }
 
+    toEcef(altitude: number): [number, number, number] {
+        const globeMetersToEcef = mercatorZfromAltitude(1, 0.0) * 2.0 * globeRadius * Math.PI;
+        const altInEcef = altitude * globeMetersToEcef;
```

Comment:
This could import https://github.com/mapbox/mapbox-gl-js/blob/ad9c713939882f23db7902361d1d34a96280def6/src/geo/projection/globe_util.js#L51